src/basic/user-util.c

```diff
@@ -602,3 +604,27 @@ bool valid_home(const char *p) {
 
         return true;
 }
+
+int maybe_setgroups(size_t size, const gid_t *list) {
+        static int cached_can_setgroups = -1;
```

Comment:
hmm, i wonder if it's a good idea to cache this, after all it can change during runtime? Also, afaics we are unlikely to call this frequently anyway, hence there's no point in caching this, is there?